### ISSUE TITLE
Rename library to `vdaf`

### ISSUE THREAD

**Issue by branlwyd** [divviup/libprio-rs#450]
This repository is currently named `libprio-rs`, exporting a crate named `prio`.

This is a poor name: while `libprio-rs` does include a `Prio3` implementation, it also contains trait definitions & functionality useful for generic VDAFs, as well as several other VDAF implementations.

For that reason, we should rename this library & crate to `vdaf`.

While working on this, we should consider how this change will affect existing library users. (We probably don't want to yank previous versions, but we should publish one final version of `libprio-rs` with a big deprecation warning pointing to the new `vdaf` library. Does Rust/Cargo provide any deprecation-warning functionality?)

**Comment by divergentdave** [comment]
The RustCrypto organization has handled name changes a few times, here's some artifacts from how they've done it.

https://docs.rs/crate/aes-ctr/0.99.99
https://sourcegraph.com/crates/aes-ctr@v0.99.99/-/blob/src/lib.rs
https://rustsec.org/advisories/RUSTSEC-2021-0061.html

**Comment by cjpatton** [comment]
Renaming is a good idea. Would it be desirable to have a more distinctive name? (Suppose someone else wanted to implement VDAFs in rust: It might not be "fair" for us to take the most generic name.)

**Comment by tgeoghegan** [comment]
It's too late for fair, I already squatted the name [`vdaf`](https://crates.io/crates/vdaf).

**Comment by divergentdave** [comment]
When we cut over, we should consider releasing the same code (except for Cargo.toml) twice, under the old and new names, to make things easy for `cargo-vet` auditors. If `vdaf 0.1.0` is nearly identical to `prio 0.LAST.0`, then that makes it easy to justify adding an exemption for the 'new' crate.

**Comment by tgeoghegan** [comment]
Yeah, if VDAF is nearly finished then this would be a good time. Though we should do the refactoring of `mod codec` into a distinct crate before this, so that consumers of `prio` can deal with all that breakage at once.
